### PR TITLE
[Network] Restructure DNS Commands

### LINKED ISSUE #1134: [DNS] Issues with adding/removing DNS records in new CLI 2.0
DNS record add/remove has currently been implemented via:

az network dns record <record type> add
az network dns record <record type> remove

The help at 'az network dns record -h' is very cryptic and unhelpful, it has no explanation of what the user is supposed to do.

Specifying the record type as above is confusing since in other commands (record-set commands) the type is specified via a parameter (--type).  It's confusing to support type in two different ways.

It is not possible to create a record set and a record in a single command.  That should be possible and would be very useful.  Likewise to remove.

I think we need to meet discuss the options here, work out the best way forward.

### LINKED ISSUE #2087: [DNS] Change 'az network dns record update-soa' to 'az network dns record soa update'
Under 'az network dns record', we have a bunch of subgroups (one for each record type) followed by a one-off command 'update-soa'.

The SOA record set is a bit of a special case, in that it can be edited but doesn't support add/remove.

However, I think there's a cleaner way to support this. Remove the 'update-soa' command, and move it to an 'update' command under the 'soa' command group

That way, under 'records' we have a clean list of command groups, one for each record type including SOA. Under each record type, we support add/remove commands, except for SOA which supports an update command instead.

### PR DESCRIPTION
**Pull request by tjprescott**
Fixes #2087. Fixes #1134.

Restructures the DNS command tree. cc/ @jtuliani

**Breaking Changes**
- The `record` subgroup has been removed.
- A subgroup for each record type has been added to the `record-set` subgroup.
- The add/remove commands that were previously in `record` are now listed in `record-type <type>` as `add-record`/`remove-record`.
- A new command `record-set list` was added that lists all record sets in the zone. This was previously triggered by an optional parameter.
- The show/delete/update/etc. commands for each type of record set no longer require the type parameter. It is determined behind the scenes.
- The `soa` subgroup now has two commands: show and update. Name is no longer required (it is always '@').
- The `add` command for CNAME records has been renamed `set-record` because there can only be one value in a given CNAME record set.

### CODE REVIEW

**Comment by tjprescott** [comment]
**az network dns -h**
```
Group
    az network dns: Host your DNS domain in Azure.

Subgroups:
    record-set: Manage DNS records and record sets.
    zone      : Manage DNS zones.
```

**az network dns record-set -h**
```
Group
    az network dns record-set: Manage DNS records and record sets.

Subgroups:
    a    : Manage DNS A records.
    aaaa : Manage DNS AAAA records.
    cname: Manage DNS CNAME records.
    mx   : Manage DNS MX records.
    ns   : Manage DNS NS records.
    ptr  : Manage DNS PTR records.
    soa  : Manage DNS zone SOA record.
    srv  : Manage DNS SRV records.
    txt  : Manage DNS TXT records.

Commands:
    list : List all record sets within a DNS zone.
```

**az network dns record-set a -h**
```
Group
    az network dns record-set a: Manage DNS A records.

Commands:
    add-record   : Add A record.
    create       : Create an empty A record set.
    delete       : Delete a A record set and all records within.
    list         : List all A record sets in a zone.
    remove-record: Remove A record from its record set.
    show         : Show details of A record set.
    update       : Update A record set.
```

**az network dns record-set soa -h**
```
Group
    az network dns record-set soa: Manage DNS zone SOA record.

Commands:
    show  : Show details of the DNS zone's SOA record.
    update: Update properties of the zone's SOA record.
```

File: src/command_modules/azure-cli-network/azure/cli/command_modules/network/_params.py [hunk 1]
```diff
@@ -557,46 +555,42 @@ def completer(prefix, action, parsed_args, **kwargs): # pylint: disable=unused-a
...
+register_cli_argument('network dns record-set soa', 'refresh_time', options_list=('--refresh-time', '-f'), help='Refresh value (seconds).')
+register_cli_argument('network dns record-set soa', 'retry_time', options_list=('--retry-time', '-r'), help='Retry time (seconds).')
+register_cli_argument('network dns record-set soa', 'serial_number', options_list=('--serial-number', '-s'), help='Serial number.')
+register_cli_argument('network dns record-set srv', 'priority', options_list=('--priority', '-p'), help='Priority metric.')
+register_cli_argument('network dns record-set srv', 'weight', options_list=('--weight', '-w'), help='Weight metric.')
+register_cli_argument('network dns record-set srv', 'port', options_list=('--port', '-r'), help='Service port.')
+register_cli_argument('network dns record-set srv', 'target', options_list=('--target', '-t'), help='Target domain name.')
+register_cli_argument('network dns record-set txt', 'value', options_list=('--value', '-v'), nargs='+', help='Space separated list of text values which will be concatenated together.')
```
Comment (derekbekoe) [inline src/command_modules/azure-cli-network/azure/cli/command_modules/network/_params.py]: Will it be concatenated to include the space or not?
Why `nargs='+'` if I can just pass in a single string that has my own spaces in?

File: src/command_modules/azure-cli-network/azure/cli/command_modules/network/tests/test_network_commands.py [hunk 2]
```diff
@@ -1177,7 +1177,7 @@ def body(self):
 class NetworkDnsScenarioTest(ResourceGroupVCRTestBase):
 
     def __init__(self, test_method):
-        super(NetworkDnsScenarioTest, self).__init__(__file__, test_method, resource_group='cli_test_dns')
+        super(NetworkDnsScenarioTest, self).__init__(__file__, test_method, resource_group='cli_test_dns', debug=True)
```
Comment (derekbekoe) [inline src/command_modules/azure-cli-network/azure/cli/command_modules/network/tests/test_network_commands.py]: Remove `debug=True`

File: src/command_modules/azure-cli-network/azure/cli/command_modules/network/_params.py (hunk 1 above)
Comment (tjprescott) [inline src/command_modules/azure-cli-network/azure/cli/command_modules/network/_params.py]: Single string with spaces will preserver the spaces. Space separated list (no quotes) will concatenate the strings. It's a DNS thing.

**Comment by tjprescott** [comment]
@derekbekoe debug=False 💃

**Pull request merged**